From 24f3d30cdec3c7473762405bde1c85b7c26f7941 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Tue, 05 Feb 2019 10:20:55 +0200
Subject: [PATCH] GCC: use Wno-error=header-hygiene only with clang.

There is no sense in passing -Wno-error=header-hygiene
if -Wno-header-hygiene is not passed (gcc case). Thus,
use -Wno-error=header-hygiene only with clang.

Bug: chromium:819294
Change-Id: I01bd9771352ab2e8827f1fdd493eb89bb719dd0f
Reviewed-on: https://swiftshader-review.googlesource.com/c/24428
Kokoro-Presubmit: kokoro <noreply+kokoro@google.com>
Reviewed-by: Alexis Hétu <sugoi@google.com>
Tested-by: Alexis Hétu <sugoi@google.com>
---

diff --git a/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn b/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
index 4ccc6d1..0f1b937 100644
--- a/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
+++ b/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
@@ -55,7 +55,9 @@
   }
 
   if (is_clang) {
-    cflags += [ "-Wno-header-hygiene" ]
+    cflags += [ "-Wno-header-hygiene",
+                "-Wno-error=header-hygiene"
+    ]
     if (target_cpu == "x86" || target_cpu == "x64") {
       cflags += [ "-msse2" ]
     }
@@ -70,7 +72,6 @@
     "-Wno-unused-private-field",
     "-Wno-unused-result",
     "-Wno-unused-variable",
-    "-Wno-error=header-hygiene",
   ]
 
   defines = [
